internal/datatransfer/monitor.go

```diff
@@ -112,52 +121,98 @@ func (m *Monitor) getPathsOfWorkload(workloadName string) []*models.DataPath {
 	return dataPaths
 }
 
+func (m *Monitor) ForceSync() error {
+	return m.syncPaths()
+}
+
+func (m *Monitor) HasStorageDefined() bool {
+	storage := m.config.GetDeviceConfiguration().Storage
+	if storage == nil {
+		return false
+	}
+
+	return storage.S3 != nil
+}
+
+func (m *Monitor) InitStorage() error {
+	storage := m.config.GetDeviceConfiguration().Storage
```

Comment:
When the device worker is started for the first time and `Monitor` is created, `m.config.GetDeviceConfiguration().Storage` will most likely be nil and the `fsSync` will never be configured correctly. At least until device-worker is restarted.